### PR TITLE
[WFLY-15748] / [WFLY-13889] Remove PicketBox from the testsuite.

### PR DESCRIPTION
**Pull request by darranl**
This supersedes https://github.com/wildfly/wildfly/pull/14949 as GitHub was incorrectly reporting a merge conflict.

Remove dependencies on the PicketBox module and APIs for the testsuite.
https://issues.redhat.com/browse/WFLY-15748
Also resolve https://issues.redhat.com/browse/WFLY-15843

### CODE REVIEW

**Comment by darranl** [comment]
@ropalka / @bstansberry This one is ready to review to get this batch of testsuite deletions in so we can see what is left.

**Review by ropalka — CHANGES_REQUESTED** [review]
Wouldn't it be possible to rework this PR to simplify and ensure proper review @darranl ?

It was easy to review commits related to Picketbox removal from test suite,
but the last three "sync with upstream" are almost impossible to review :(
I can't verify them if they are correct or if there is something missing.

My proposal is:
 * don't use git merge
 * start a new branch from current upstream
 *  cherry-pick commit by commit into new branch

**Comment by darranl** [comment]
TBH on this one the "Files Changed" tab is giving an accurate view on this overall PR.

**Review by ropalka — APPROVED** [review]
Thanks @darranl for the hint how to approach this PR review. I was used to review PRs commit by commit and didn't use "Files Changed" till now. I am approving this PR. All test changes looks good.

**Comment by darranl** [comment]
@bstansberry could you please take a look at this PR.

After removing the dependency on PicketBox I was getting ts.bootable test failures but looking more closely these were in some redundant legacy-security so I have removed those as no longer applicable.

File: testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/formauth/SecureServlet.java [hunk 1]
```diff
@@ -45,13 +43,13 @@
...
-            if (subject == null)
-                throw new ServletException("No valid subject found, user=" + user);
-        }
+//        String validateSubject = request.getParameter("validateSubject");
+//        if (validateSubject != null && Boolean.valueOf(validateSubject).booleanValue()) {
+//            // Assert that there is a valid SecurityAssociation Subject
+//            Subject subject = SecurityContextAssociation.getSubject();
+//            if (subject == null)
+//                throw new ServletException("No valid subject found, user=" + user);
+//        }
```
Comment (bstansberry) [inline testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/formauth/SecureServlet.java:52]: If you make other changes to this PR these lines can be deleted and the commit message changed to 'Remove unused verification dependent on Picketbox."

Both in main and in downstream there's no code that passes requrest parameter "validateSubject" so this is all cruft.

File: testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/security/cert/WebSecurityCERTTestCase.java [hunk 2]
```diff
@@ -68,6 +53,7 @@
 @RunAsClient
 @ServerSetup(WebCERTTestsSetup.class)
 @Category(CommonCriteria.class)
+@Ignore("[WFLY-15749] Update WebSecurityCERTTestCase to use Elytron SSLContext.")
```
Comment (bstansberry) [inline testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/security/cert/WebSecurityCERTTestCase.java:56]: I made this a Blocker.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/callerprincipal/GetCallerPrincipalTestCase.java [hunk 3]
```diff
@@ -215,7 +215,8 @@ public void testStatelessLifecycle() throws Exception {
             Assert.assertEquals(OK + "stop", results.getSlsb("predestroy"));
             return null;
         };
-        Util.switchIdentitySCF("user1", "password1", callable);
+        //Util.switchIdentitySCF("user1", "password1", callable);
+        callable.call();
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/callerprincipal/GetCallerPrincipalTestCase.java:219]: Unless I'm missing something the callable should just be removed and its code inlined.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/callerprincipal/GetCallerPrincipalTestCase.java [hunk 4]
```diff
@@ -234,7 +235,8 @@ public void testStatefulLifecycle() throws Exception {
             return null;
         };
         try {
-            Util.switchIdentitySCF("user1", "password1", callable);
+            //Util.switchIdentitySCF("user1", "password1", callable);
+            callable.call();
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/callerprincipal/GetCallerPrincipalTestCase.java:239]: Unless I'm missing something the callable should just be removed and its code inlined.

**Review by bstansberry — CHANGES_REQUESTED** [review]
@darranl I made a minor change request, to help the GetCallerPrincipalTestCaseclass be more understandable in the future.  Ignoring another test isn't great but it's ok.

**Comment by darranl** [comment]
@bstansberry I have just pushed again with the requested changes.

**Review by bstansberry — APPROVED** [review]

**Pull request merged**